### PR TITLE
Under 13 flow

### PR DESCRIPTION
**Pull request by georgyangelov**
Updates to show an under-13-specific ux flow for users when registering.

See also:

- https://github.com/scratchfoundation/scratchr2/pull/6623
- https://github.com/scratchfoundation/scratch-api/pull/1873

### CODE REVIEW

File: src/components/dropdown-banner/email-confirmation/banner.jsx [hunk 1]
```diff
@@ -24,7 +25,11 @@ const EmailConfirmationBanner = ({onRequestDismiss}) => {
                 onRequestDismiss={onRequestDismiss}
             >
                 <FormattedMessage
-                    id="emailConfirmationBanner.confirm"
+                    id={
+                        userUsesParentEmail ?
+                            "emailConfirmationBanner.under13.confirm" :
```
[the remark this answers is not in the document]
Comment (georgyangelov) [reply src/components/dropdown-banner/email-confirmation/banner.jsx:30]: Good point, I also noticed that yesterday, will update it to be more consistent. It was originally under 13 everywhere but "uses parent email" was more descriptive so I changed it

File: src/components/join-flow/join-flow.jsx [hunk 2]
```diff
@@ -229,6 +230,39 @@ class JoinFlow extends React.Component {
...
+
+        if (yearDiff > 13) {
+            return false;
+        } else if (yearDiff < 13) {
+            return true;
+        } else {
+            const currentMonth1Based = now.getMonth() + 1;
+            const monthsLeftToBirthday = birthMonth - currentMonth1Based;
+
+            return monthsLeftToBirthday >= 0;
```
[the remark this answers is not in the document]
Comment (georgyangelov) [reply src/components/join-flow/join-flow.jsx:262]: You should be considered 12, this is what the logic does - in this case `monthsLeftToBirthday` would be `0`, so the function will return `true`

[the remark this answers is not in the document]
Comment (cwillisf) [reply src/components/join-flow/join-flow.jsx:262]: Ah! Right, sorry, somehow when I looked at the `=` in `>=`, I momentarily flipped the logic in my head. Carry on then :sweat_smile:

[the remark this answers is not in the document]
Comment (georgyangelov) [reply src/components/join-flow/join-flow.jsx:262]: I initially thought about it for a couple of minutes, wrote it, tested and it turned out I had it the other way around. 😄

**Comment by mrsrec** [comment]
@cwillisf The 2.0 registration page says that anyone under **16** should enter a parent's email, not just 13. This was changed in 2018, possibly due to GDPR? I'm not sure, if it makes a difference, just seemed like it's possibly relevant.

File: src/components/dropdown-banner/email-confirmation/banner.jsx [hunk 3]
```diff
@@ -8,8 +8,12 @@ const FormattedMessage = require('react-intl').FormattedMessage;
 const EmailConfirmationModal = require('../../../components/modal/email-confirmation/modal.jsx');
 
 const EmailConfirmationBanner = ({onRequestDismiss, userUsesParentEmail}) => {
-
     const [showEmailConfirmationModal, setShowEmailConfirmationModal] = useState(false);
+    const i18nPrefix =
+        userUsesParentEmail ?
+            'emailConfirmationBanner.parentEmail' :
+            'emailConfirmationBanner';
```
Comment (cwillisf) [inline src/components/dropdown-banner/email-confirmation/banner.jsx:15]: Nice!

**Review by cwillisf — APPROVED** [review]

**Pull request merged**